collector/xml-eater.py

```diff
@@ -0,0 +1,48 @@
+from lxml import etree 
+import numpy as np
+import re
+#xml_file: file to be parsed
+#sample_rate: sample rate of audio that goes with xml_file
+#deliminator: deliminator for speaker labels.
+#             e.g. if we have "[M] Something something talking blah blah", 
+#                  then deliminator is a regex "\[[A-Z]\]". (match open square bracket, any single letter, close square bracket.)
+#discard: discard any lines that contain this regex (e.g., if they label sound effects in a consistant way, we can get rid of 
+#		those pretty easily using this param.)                      
+def xml_eater(xml_file, sample_rate, deliminator, discard=None):
+	# open xml
+	xml_file = open(xml_file).read()
+	tree = etree.fromstring(xml_file)	
+	starts = tree.xpath("text/@start")
+	texts  = [x.text.strip("\n") for x in tree.xpath("text")]
+	durs   = tree.xpath("text/@dur") 
+	speakers = [] # list of all possible speakers, used to determine size and entries of np array
+	# fill speakers
+	for text in texts:
+		for speaker in re.findall(deliminator, text):
+			if speaker not in speakers: speakers.append(speaker)
+	arr = np.zeros( (1, len(speakers)), dtype=np.int64)
+	current = None	
+	n = 0
+	# BUG: doesn't count time in between captions; TODO: fix this
+	while n < len(texts):
+		if check_valid(texts[n], deliminator, discard): 
+			try: current = re.match(deliminator, texts[n]).group()
+			except AttributeError: pass
+		else:
+			current = None
+		newrow = [int(x==current) for x in speakers]
+		start = float(starts[n])
+		dur = float(durs[n])
+		for i in range(int(start*100), int((start+dur)*100)):
```

Comment:
Where is this 100 coming from?

Reply:
Everything is given in seconds up to 2 decimal places in the xml file, so I've multiplied everything by 100 so we've got integers.  That also means I've gone slightly away from what the issue said to do. (Basically just replace "seconds" by "hundredths of seconds" in the issue.)